static/js/recent_topics.js

```diff
@@ -59,4 +59,17 @@ exports.get = function () {
     return get_sorted_topics();
 };
 
+exports.process_topic_edit = function (old_stream_id, old_topic, new_topic, new_stream_id) {
+    // See `recent_senders.process_topic_edit` for
+    // logic behind this and important notes on use of this function.
+    topics.delete(old_stream_id + ':' + old_topic);
+
+    const old_topic_msgs = message_util.get_messages_in_topic(old_stream_id, old_topic);
+    exports.process_messages(old_topic_msgs);
+
+    new_stream_id = new_stream_id || old_stream_id;
+    const new_topic_msgs = message_util.get_messages_in_topic(new_stream_id, new_topic);
+    exports.process_messages(new_topic_msgs);
```

Comment:
4 calls to `message_util.get_messages_in_topic` is starting to get expensive; we may end up feeling the need to profile this to see if it's problematic.  

I think basically I just want to profile doing something very fast in `message_util.get_messages_in_topic` in a production environment with 50K unread messages; remind me to do that (I have a few easy places to test).

Reply:
Yes, this can delay topic & stream edit updates significantly. The solution would be to store a dictionary of message ids per topic and maintain it, which will be less costly.

Reply:
Well, can it actually?  We need to test.  It's very possible that `message_util.get_messages_in_topic` takes 2ms to run.

Reply:
OK, so I ran `message_util.get_messages_in_topic` with 10,000 messages fetched. (I went to the beginning of the messages). It takes around 1.5ms to execute per call. So, this isn't that expensive. (It looked like it was)

Reply:
Great, thanks for checking!